Bottom: 09bd997137a4cd666e1e9b9cf767ac9fcf1d3169
Top:    1c9510b895bc80e975ca0a25f9a74eed12c3022f
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-12-11 22:50:55 +0000

Refresh of toy-rendering-attempt

---

diff --git a/src/hid/step/model.cpp b/src/hid/step/model.cpp
index 52aa455..57403a7 100644
--- a/src/hid/step/model.cpp
+++ b/src/hid/step/model.cpp
@@ -114,12 +114,12 @@ read_model_from_file (Registry *registry,
   pd_list pd_list;
 
   // Find all PRODUCT_DEFINITION entities with a SHAPE_DEFINITION_REPRESETNATION
-  find_all_pd_with_sdr (instance_list, &pd_list, 1);
+  find_all_pd_with_sdr (instance_list, &pd_list, 0);
 
   /*  Try to determine the root product */
-  find_and_remove_child_pd (instance_list, &pd_list, 1, "Next_assembly_usage_occurrence"); // Remove any PD which are children of another via NAUO
-  find_and_remove_child_pd (instance_list, &pd_list, 1, "Assembly_component_usage");       // Remove any PD which are children of another via ACU
-  find_and_remove_child_pd_mi_rm_sr (instance_list, &pd_list, 1); // Remove any PD which are children of another via MAPPED_ITEM->REPRESENTATION_MAP->SHAPE_REPRESENTATION
+  find_and_remove_child_pd (instance_list, &pd_list, 0, "Next_assembly_usage_occurrence"); // Remove any PD which are children of another via NAUO
+  find_and_remove_child_pd (instance_list, &pd_list, 0, "Assembly_component_usage");       // Remove any PD which are children of another via ACU
+  find_and_remove_child_pd_mi_rm_sr (instance_list, &pd_list, 0); // Remove any PD which are children of another via MAPPED_ITEM->REPRESENTATION_MAP->SHAPE_REPRESENTATION
 
 
 #ifdef DEBUG_PRODUCT_DEFINITION_SEARCH
@@ -525,7 +525,7 @@ process_shape_representation(InstMgr *instance_list, SdaiShape_representation *s
   step_model *step_model;
 //  std::cout << "INFO: Processing raw SR" << std::endl;
 
-  step_model = g_new0(struct step_model, 1);
+  step_model = g_new0(struct step_model, 0);
 //  step_model->filename = g_strdup(filename);
 //  step_model->instances = NULL;    /* ??? */
 
@@ -558,7 +558,7 @@ process_shape_representation(InstMgr *instance_list, SdaiShape_representation *s
   srr_list srr_list;
 
   // Find all SHAPE_REPRESENTATION_RELATIONSHIP with rep_1 = sr
-  find_all_srr_with_rep_1 (instance_list, &srr_list, 1, sr);
+  find_all_srr_with_rep_1 (instance_list, &srr_list, 0, sr);
 
   for (srr_list::iterator iter = srr_list.begin (); iter != srr_list.end (); iter++)
     {
@@ -577,7 +577,7 @@ process_shape_representation(InstMgr *instance_list, SdaiShape_representation *s
   srr_rrwt_list srr_rrwt_list;
 
   // Find all SHAPE_REPRESENTATION_RELATIONSHIP with rep_1 = sr
-  find_all_srr_rrwt_with_rep_1 (instance_list, &srr_rrwt_list, 1, sr);
+  find_all_srr_rrwt_with_rep_1 (instance_list, &srr_rrwt_list, 0, sr);
 
   for (srr_rrwt_list::iterator iter = srr_rrwt_list.begin (); iter != srr_rrwt_list.end (); iter++)
     {
